server/test/Mutation.test.ts

```diff
@@ -62,54 +56,34 @@ test("Octave rule works", () => {
 
   // Stays within Octave
   let set = ["D3", "G3", "A4", "B4", "D4", "G4", "A5", "G5"].map((s) => Note.fromString(s));
-  let expectedOutput = ["G3", "A4", "B4", "D4", "G4", "A5", "G5"];
+  let expectedOutput = ["G3", "A4", "B4", "D4", "G4", "G5"];
   let output = rule.apply(2, set, bot.melody).map((n) => n.note);
   expect(output.length).toBe(expectedOutput.length);
   expect(output.filter((o) => !expectedOutput.includes(o)).length === 0).toBe(true);
 });
 
-test("GA octaves", () => {
-  let worker = new GaWorker();
-
-  let highNotes = 0;
-  let totalNotes = 0;
-  let lowNotes = 0;
-  // Represents tenor range
-  let high = Note.fromString("C5");
-  let low = Note.fromString("C3");
-
-  for (var i = 0; i < 100; i++) {
-    let newBots = worker.generateNewBots(worker.initialBots());
-
-    newBots.forEach((bot) => {
-      bot.melody.notes.forEach((note) => {
-        totalNotes++;
-
-        if (high.compare(note) <= 0) {
-          highNotes++;
-        }
-
-        if (low.compare(note) >= 0) {
-          lowNotes++;
-        }
-      });
-    });
-  }
-
-  console.log(`results: high - ${highNotes} , low - ${lowNotes} , total - ${totalNotes}`);
-  console.log(`fraction high notes: ${highNotes / totalNotes}`);
-  console.log(`fraction low notes: ${lowNotes / totalNotes}`);
-});
-
 test("Vocal Range Rules", () => {
   let bot = new Bot(0, Melody.fromString("C4,G4,D4,A4"));
 
   let rule = new CounterTenorRule();
   let set = ["E6", "F5", "E5", "D5", "A3", "E3", "F3"].map((s) => Note.fromString(s));
 
   let output = rule.apply(0, set, bot.melody).map((n) => n.note);
-  let expectedOutput = ["E5", "D5", "E3", "F3"];
+  let expectedOutput = ["E5", "D5", "E3", "A3", "F3"];
 
   expect(output.length).toBe(expectedOutput.length);
   expect(output.filter((o) => !expectedOutput.includes(o)).length === 0).toBe(true);
 });
+
+test("End To End No Crash", () => {
+  let worker = new GaWorker();
+
+  // create set of bots
+  let bots = worker.initialBots();
+  bots.forEach(bot => bot.metric = Math.floor(Math.random() + 1));
+
+  worker.generateNewBots(bots);
+
+  // pass if no crash
+  expect(true).toBe(true);
```

Comment:
No need for this last assertion, the test will be reported as having passed either way.

Reply:
good catch! will fix.